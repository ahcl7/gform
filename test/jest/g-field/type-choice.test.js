import { generateComponent } from "./common";

describe('GField with type choice', function () {
  it('should render correctly', () => {
    const component = generateComponent({
      key: 'layout', type: 'choice', choiceKey: 'layoutType',
      fields: [{
        key: 'linear', type: 'object',
        fields: [{ key: 'linearName', type: 'input', flex: 'md-12' }]
      }, {
        key: 'grid', type: 'object',
        fields: [{ key: 'gridName', type: 'input', flex: 'md-12' }]
      }]
    });
    expect(component.html()).toMatchSnapshot();
  });
});
