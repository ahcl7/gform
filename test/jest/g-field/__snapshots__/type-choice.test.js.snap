// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GField with type choice should render correctly 1`] = `
<g-col-stub xs12>
  <g-menu-stub offset-y
               z-index="1000"
               modelvalue="false"
               closeoncontentclick="true"
  >
    <g-list-stub items="[object Object],[object Object]">
    </g-list-stub>
  </g-menu-stub>
</g-col-stub>
`;
